events/repository.js

```diff
@@ -0,0 +1,78 @@
+const path = require('path');
+const fs = require('fs');
+const { readFile, writeFile, mapRowOnHeaderTransformStream } = require('./csv_file_helper');
+
+/* Events Schema
+* id
+* title
+* description
+* location
+* timestamp
+*/
+
+const eventsFilePath = path.join('.', 'events', 'events.csv')
+
+const EventsRepository = {
+  batchFind: () => {
+    return fs.createReadStream(eventsFilePath).pipe(mapRowOnHeaderTransformStream);
+  },
+  find: async (location) => {
+    try {
+      const events = await readFile(eventsFilePath);
+      if (!location || location === 'all') {
+        return events;
+      }
+      return events.filter((event) => event.location === location);
+    } catch (err) {
+      throw err;
+    }
+  },
+  findOne: async (id) => {
+    try {
+      const events = await readFile(eventsFilePath);
+      return events.find((event) => event.id === id);
```

Comment:
Potentially we could avoid loading all the events into the memory by filtering needed events while reading data from the file (using streams). The node process would use less memory.